libs/algo/odc/algo/io.py

```diff
@@ -84,17 +86,30 @@ def _load_with_native_transform_1(
 
     (ds,) = sources.data[0]
     load_geobox = compute_native_load_geobox(geobox, ds, basis)
+
     if pad is not None:
         load_geobox = gbox.pad(load_geobox, pad)
 
     mm = ds.type.lookup_measurements(bands)
     xx = Datacube.load_data(sources, load_geobox, mm, dask_chunks=load_chunks)
-    xx = native_transform(xx)
 
+
+    xxs = native_transform(xx)
```

Comment:
`native_transform` is to return a `xr.Dataset`, all the operations in this function are based on this assumption. Even if the mask bands need to be merged here (which I don't agree, but it's separate), they don't have to be in a list but as variables in the same `Dataset`.